### DISCUSSION TITLE
INMP441 channel & bit conversion

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#800] [Q&A] [answered]
I have INMP441 MCROPHONE WHOSE L/R PIN IS CONNECTD TO GROUND .   

IT WORK WITH 2 CHANNEL AND SEND DATA TO SBC for further processing  but data size is large due to 2 channels so i  add channel from 2 to 1 in format converter stream which i was using for 32 bit to 16 bit conversion only and converter is working with this .  
 
but after adding  channel 2 to 1 in FormatConverterStream then i get noise only . 

what think  to consider for channel conversion  ?
From channel 2 to 1 which channel of INMP441 will it use ?  should i connect L/R to high in order to select the correct channel ? 

AudioInfo from(44100,2,32);
AudioInfo to  (44100,1,16);
FormatConverterStream conv(out);
EncodedAudioStream encStream(out, enc);
.
conv.begin(from, to);

**Comment by pschatzmann** [comment, marked as answer]
Strange, I am pretty sure that I have tested this scenario.
If it is not working you can still use a ChannelFormatConverterStream and a NumberFormatConverterStream

**Comment by aqildad-create** [comment]
L/R pin is grounded then after ChannelFormatConverterStream  from 2 to 1 which channel will eliminated ? channel 1 or 2 ? or else converter knows the channel which values is zeros  will be eliminated ?

**Reply by pschatzmann** [reply]
No channel is eliminated: they are mixed together from 2 to 1.
